<a href="https://colab.research.google.com/github/KacperWardzala/data-science-bootcamp2023/blob/main/Project_Exploring_eBay_Car_Sales_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Project: Exploring eBay Car Sales Data

#### Downloading and preaparing data

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
url = 'https://query.data.world/s/2t76lim4udd74b7iee7qp2xeisrgxs?dws=00000'
autos = pd.read_csv(url, encoding='latin-1')

In [ ]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

#### About project


Dataset contains over 370000 used cars from Ebay-Kleinanzeigen.

The oryginal content of the data is in german. Coding in this csv file have to be changed to latin-1. File contains 20 columns mostly filled with strings. he column names use camelcase instead of snakecase. Fields are included in the file:

* dateCrawled : when this ad was first crawled, all field-values are taken from this date
* name: "name" of the car
* seller: private or dealer
* offerType: The type of listing
* price: the price on the ad to sell the car
* abtest: Whether the listing is included in an A/B test
* vehicleType: The vehicle Type
* yearOfRegistration: at which year the car was first registered
* gearbox
* powerPS: power of the car in PS
* model: car model name
* kilometer: how many kilometers the car has driven
* monthOfRegistration: at which month the car was first registered
* fuelType: type of fuel used by cars
* brand: name of car's brand
* notRepairedDamage: if the car has a damage which is not repaired yet
* dateCreated: the date for which the ad at ebay was created
* nrOfPictures: number of pictures in the ad
* postalCode
* lastSeenOnline: when the crawler saw this ad last online

#### Names of Columns

In [ ]:
columns = autos.columns
columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [ ]:
columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
    'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
    'kilometer', 'registration_month', 'fuel_type', 'brand',
    'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
    'last_seen']


In [ ]:
autos.columns = columns

In [ ]:
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price abtest vehicle_type  registration_year    gearbox  power_ps  model  \
0    480   test          NaN               1993    manuell         0   golf   
1  18300   test        coupe               2011    manuell       190    NaN   
2   9800   test          suv               2004  automatik       163  grand   
3   1500   test   kleinwagen               2001    manuell        75   golf   
4   3600   test   kleinwagen               2008    manuell        69  fabia   

   kilometer  registration_month fuel_type       brand unrepaired_damage  \
0     150000                   0    benzin  volkswagen               NaN   
1     125000                   5    diesel        audi                ja   
2     125000                   8    diesel        jeep               NaN   
3     150000                   6    benzin  volkswagen              nein   
4      90000                   7    diesel       skoda              nein   

            ad_created  nr_of_pictures  postal_code            last_seen  
0  2016-03-24 00:00:00               0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00               0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00               0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00               0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00               0        60437  2016-04-06 10:17:21

I changed all column names to the snake_case from the camelCase by creating a new list with correct names.

#### Initial Exploration and Cleaning

In [ ]:
autos.describe()

price  registration_year       power_ps      kilometer  \
count  3.715280e+05      371528.000000  371528.000000  371528.000000   
mean   1.729514e+04        2004.577997     115.549477  125618.688228   
std    3.587954e+06          92.866598     192.139578   40112.337051   
min    0.000000e+00        1000.000000       0.000000    5000.000000   
25%    1.150000e+03        1999.000000      70.000000  125000.000000   
50%    2.950000e+03        2003.000000     105.000000  150000.000000   
75%    7.200000e+03        2008.000000     150.000000  150000.000000   
max    2.147484e+09        9999.000000   20000.000000  150000.000000   

       registration_month  nr_of_pictures   postal_code  
count       371528.000000        371528.0  371528.00000  
mean             5.734445             0.0   50820.66764  
std              3.712412             0.0   25799.08247  
min              0.000000             0.0    1067.00000  
25%              3.000000             0.0   30459.00000  
50%              6.000000             0.0   49610.00000  
75%              9.000000             0.0   71546.00000  
max             12.000000             0.0   99998.00000

In [ ]:
autos.describe(include='all')

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

        abtest vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  nr_of_pictures   postal_code  \
count             299468               371528        371528.0  371528.00000   
unique                 2                  114             NaN           NaN   
top                 nein  2016-04-03 00:00:00             NaN           NaN   
freq              263182                14450             NaN           NaN   
mean                 NaN                  NaN             0.0   50820.66764   
std                  NaN                  NaN             0.0   25799.08247   
min                  NaN                  NaN             0.0    1067.00000   
25%                  NaN                  NaN             0.0   30459.00000   
50%                  NaN                  NaN             0.0   49610.00000   
75%                  NaN                  NaN             0.0   71546.00000   
max                  NaN                  NaN             0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                  

* The *pictures_number* column is empty
* The minimum in the *registration_month* column is 0 which is not possible. This also suggests inaccurate rows of data
* The *seller* and the *offer_type* columns contain very simillar values, so one of them can be removed
* Dataset has been changed, and the column *odometer* is changed to the *kilometer* and they have number values


In [ ]:
price_and_kilometer = autos[['price', 'kilometer']]
price_and_kilometer.head()

price  kilometer
0    480     150000
1  18300     125000
2   9800     125000
3   1500     150000
4   3600      90000

#### Exploring the Odometer and Price Columns

##### Number of unique values

Price column:

In [ ]:
price_and_kilometer['price'].unique().shape


(5597,)

In [ ]:
price_and_kilometer['price'].describe()

count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64

In [ ]:
price_count = price_and_kilometer['price'].value_counts()
top_price_count = price_count.sort_index(ascending=False)
top_price_count.head(15)

2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
Name: price, dtype: int64

Here we can see that there is a jump up from 3895000 to 9999999 dollars. Those prices looks too high for car prices. So there I should remove data that is greater than 3895000$.

In [ ]:
price_count = price_and_kilometer['price'].value_counts()
notop_price_count = price_count.sort_index(ascending=True)
notop_price_count.head(15)

0     10778
1      1189
2        12
3         8
4         1
5        26
7         3
8         9
9         8
10       84
11        5
12        8
13        7
14        5
15       27
Name: price, dtype: int64

There is many 0$ offerts what is not regular price for eBay. 1 dollar prices can be starting bids of auctions, so I should remove 0 dollar data.

In [ ]:
autos = autos[autos['price'].between(1, 3895001)]
autos['price'].describe()

count    3.606980e+05
mean     6.068401e+03
std      1.818741e+04
min      1.000000e+00
25%      1.250000e+03
50%      3.000000e+03
75%      7.490000e+03
max      3.895000e+06
Name: price, dtype: float64

Kilometer column:

In [ ]:
price_and_kilometer['kilometer'].unique().shape

(13,)

In [ ]:
price_and_kilometer['kilometer'].describe()

count    371528.000000
mean     125618.688228
std       40112.337051
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: kilometer, dtype: float64

In [ ]:
kilometer_count = price_and_kilometer['kilometer'].value_counts()
top_kilometer_count = kilometer_count.sort_index(ascending=True)
top_kilometer_count.head(15)

5000        7069
10000       1949
20000       5676
30000       6041
40000       6376
50000       7615
60000       8669
70000       9773
80000      11053
90000      12523
100000     15920
125000     38067
150000    240797
Name: kilometer, dtype: int64

In [ ]:
kilometer_count = price_and_kilometer['kilometer'].value_counts()
top_kilometer_count = kilometer_count.sort_index(ascending=False)
top_kilometer_count.head(15)

150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
40000       6376
30000       6041
20000       5676
10000       1949
5000        7069
Name: kilometer, dtype: int64

Rows in kilometer column looks correct and there I cannot see any outliers.

#### Exploring the date columns

There are a number of columns that represent date values:

* date_crawled
* last_seen
* ad_created
* registration_month
* registration_year

In [ ]:
autos[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

In [ ]:
autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.025545
2016-03-06    0.014480
2016-03-07    0.035653
2016-03-08    0.033466
2016-03-09    0.034112
2016-03-10    0.032648
2016-03-11    0.032775
2016-03-12    0.036244
2016-03-13    0.015781
2016-03-14    0.036327
2016-03-15    0.033424
2016-03-16    0.030200
2016-03-17    0.031655
2016-03-18    0.013119
2016-03-19    0.035268
2016-03-20    0.036399
2016-03-21    0.035695
2016-03-22    0.032493
2016-03-23    0.032002
2016-03-24    0.029914
2016-03-25    0.032798
2016-03-26    0.031969
2016-03-27    0.030230
2016-03-28    0.035060
2016-03-29    0.034123
2016-03-30    0.033529
2016-03-31    0.031874
2016-04-01    0.034150
2016-04-02    0.035096
2016-04-03    0.038811
2016-04-04    0.037630
2016-04-05    0.012786
2016-04-06    0.003127
2016-04-07    0.001619
Name: date_crawled, dtype: float64

There I cannot see any outliers in data.

In [ ]:
autos['ad_created'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2014-03-10    0.000003
2015-03-20    0.000003
2015-06-11    0.000003
2015-06-18    0.000003
2015-08-07    0.000003
                ...   
2016-04-03    0.038999
2016-04-04    0.037738
2016-04-05    0.011619
2016-04-06    0.003119
2016-04-07    0.001555
Name: ad_created, Length: 114, dtype: float64

The majority of the ads were created around the same time as the crawling dates.

In [ ]:
autos['last_seen'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.001264
2016-03-06    0.004098
2016-03-07    0.005201
2016-03-08    0.007940
2016-03-09    0.009823
2016-03-10    0.011458
2016-03-11    0.012953
2016-03-12    0.023241
2016-03-13    0.008409
2016-03-14    0.012174
2016-03-15    0.016327
2016-03-16    0.016415
2016-03-17    0.028703
2016-03-18    0.006892
2016-03-19    0.016332
2016-03-20    0.019889
2016-03-21    0.020031
2016-03-22    0.020510
2016-03-23    0.018018
2016-03-24    0.019163
2016-03-25    0.018997
2016-03-26    0.015955
2016-03-27    0.016720
2016-03-28    0.022190
2016-03-29    0.023285
2016-03-30    0.023721
2016-03-31    0.024239
2016-04-01    0.023898
2016-04-02    0.024971
2016-04-03    0.025320
2016-04-04    0.025531
2016-04-05    0.126968
2016-04-06    0.218942
2016-04-07    0.130422
Name: last_seen, dtype: float64

There is a noticeable increase in the distribution for the "last_seen" column over the last three days. This column indicates when an ad has been removed from the site, likely due to a sale. The significant distribution spike observed in the last 3 days of the crawler recording is probably not a result of a sudden surge in sales, as the days leading up to this period had a fairly consistent distribution.

#### Dealing with Incorrect Registration Year Data

In [ ]:
autos['registration_year'].describe()

count    360698.000000
mean       2004.451877
std          82.110038
min        1000.000000
25%        1999.000000
50%        2004.000000
75%        2008.000000
max        9999.000000
Name: registration_year, dtype: float64

According to the results from the dataset, the website includes cars registered in the range of years 1000 to 9999. Such a range is not possible, so for the accuracy of further calculations, we can exclude cars with values outside the range of 1900 to 2023.

In [ ]:
autos = autos[autos['registration_year'].between(1900, 2023)]
autos['registration_year'].value_counts(normalize = True).head(15)

2000    0.064144
1999    0.061107
2005    0.060267
2006    0.055491
2001    0.054764
2003    0.054378
2004    0.054015
2002    0.052198
2007    0.048554
1998    0.047786
2008    0.044474
2009    0.042987
1997    0.038913
2010    0.034034
2011    0.033272
Name: registration_year, dtype: float64

The analysis indicates that the most frequently registered car model years are 2000, 1999, and 2005, comprising 6.41%, 6.11%, and 6.03% of the dataset, respectively.



#### Exploring Price by Brand

In [ ]:
brands = autos['brand'].value_counts(normalize = True)
brands

volkswagen        0.213531
bmw               0.108579
opel              0.107378
mercedes_benz     0.095982
audi              0.088707
ford              0.068903
renault           0.048332
peugeot           0.029977
fiat              0.025958
seat              0.018946
skoda             0.015463
mazda             0.015365
smart             0.014350
citroen           0.014031
nissan            0.013599
toyota            0.012825
hyundai           0.009960
sonstige_autos    0.009486
mini              0.009311
volvo             0.008981
mitsubishi        0.008218
honda             0.007583
kia               0.006926
suzuki            0.006318
alfa_romeo        0.006274
porsche           0.006066
chevrolet         0.004973
chrysler          0.003828
dacia             0.002468
jeep              0.002183
daihatsu          0.002150
land_rover        0.002102
subaru            0.002080
jaguar            0.001689
daewoo            0.001478
saab              0.001439
trabant           0.001376
l

Conclusions:
* Volkswagen is the most commonly listed brand, constituting 21.35% of the dataset.
* BMW and Opel follow, with 10.86% and 10.74% representation, respectively.
* Other prominent brands include Mercedes-Benz (9.60%), Audi (8.87%), and Ford (6.89%).
* Renault, Peugeot, and Fiat also have notable presence, with shares of 4.83%, 2.99%, and 2.60%.
* Less common brands, such as Trabant, Lancia, Rover, and Lada, have minimal representation, each accounting for less than 0.2% of the dataset.
* The analysis provides valuable insights into the distribution of car brands on the platform, with Volkswagen clearly dominating the listings.

In [ ]:
most_common_brands = brands[brands > 0.05].index
most_common_brands

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')

In [ ]:
brands_mean_price = {}
for brand in most_common_brands:
    brand_only = autos[autos['brand'] == brand]
    mean_price = brand_only['price'].mean()
    brands_mean_price[brand] = int(mean_price)

brands_mean_price

{'volkswagen': 5372,
 'bmw': 8482,
 'opel': 3021,
 'mercedes_benz': 8518,
 'audi': 9041,
 'ford': 3741}

The mean prices provide insights into the average pricing trends for each brand, with Audi, BMW, and Mercedes-Benz positioned as premium brands, while Volkswagen, Ford, and Opel cater to a more budget-conscious market segment.








#### Storing Aggregate Data in a DataFrame

In [ ]:
brand_mean_mileage = {}

for brand in most_common_brands:
    brand_only = autos[autos['brand'] == brand]
    mean_mileage = brand_only['kilometer'].mean()
    brand_mean_mileage[brand] = int(mean_mileage)

brand_mean_mileage

{'volkswagen': 128663,
 'bmw': 132950,
 'opel': 128929,
 'mercedes_benz': 130704,
 'audi': 129691,
 'ford': 123916}

Combineing the following series objects into a single dataframe.

In [ ]:
mean_mileage = pd.Series(brand_mean_mileage).sort_values(ascending = False)
mean_price = pd.Series(brands_mean_price).sort_values(ascending = False)

top_brand_info = pd.DataFrame(mean_price, columns = ['mean_price'])
top_brand_info['mean_mileage'] = mean_mileage
top_brand_info

mean_price  mean_mileage
audi                 9041        129691
mercedes_benz        8518        130704
bmw                  8482        132950
volkswagen           5372        128663
ford                 3741        123916
opel                 3021        128929

In summary, the results suggest that premium brands such as Audi, Mercedes-Benz, and BMW tend to have higher prices but also higher average mileage. On the other hand, mass-market brands like Volkswagen, Ford, and Opel offer lower prices, often accompanied by lower average mileage.

#### Bonus: most common brand/model combinations (With the help of ChatGPT)

In [ ]:
common_brand_model = autos.groupby(['brand', 'model']).size().reset_index(name='count')

idx = common_brand_model.groupby('brand')['count'].idxmax()

most_popular_models = common_brand_model.loc[idx]
most_popular_models

brand       model  count
2       alfa_romeo         156    602
13            audi          a4   9997
23             bmw         3er  19978
32       chevrolet      andere    996
38        chrysler      andere    374
43         citroen      andere   1815
53           dacia       logan    320
58          daewoo       matiz    158
62        daihatsu       cuore    364
74            fiat       punto   3241
82            ford       focus   5825
93           honda       civic   1300
98         hyundai     i_reihe   1250
101         jaguar      andere    299
106           jeep       grand    281
108            kia      andere    481
117           lada        niva    126
125         lancia     ypsilon    201
130     land_rover  freelander    218
138          mazda     3_reihe   1289
148  mercedes_benz    c_klasse   8637
165           mini      cooper   2108
169     mitsubishi        colt    845
177         nissan       micra   1719
189           opel       corsa  12194
200        peugeot     2_reihe   4763
205        porsche         911    941
218        renault      twingo   4816
219          rover      andere    382
226           saab      andere    363
233           seat       ibiza   2640
240          skoda     octavia   2178
246          smart      fortwo   4297
250         subaru     impreza    223
253         suzuki      andere   1040
264         toyota       yaris   1030
265        trabant         601    360
275     volkswagen        golf  29063
293          volvo         v40    751

ChatGPT helped me here with new function 'groupby' and 'idxmax'.
[Link to chat](https://chat.openai.com/share/a2d6c481-0712-4884-ba29-9d4b1b774d81)

'in' pandas:

* groupby is a function in pandas that is used for grouping rows of a DataFrame based on some criterion.
* It splits the data into groups based on the specified columns or criteria.
* After grouping, you can perform aggregate functions on each group or apply various operations to the groups.

'idxmax' in pandas:

* idxmax is a function that returns the index of the first occurrence of the maximum value in a Series.
* It is often used in conjunction with groupby to find the index of the maximum value within each group.

These results provide an overview of the distribution of specific car models across different brands in the dataset. The Volkswagen Golf stands out as the most frequently listed model, followed by other popular models such as the BMW 3 Series (3er), Opel Corsa, and Audi A4.